### DISCUSSION TITLE
RawRuleOf<AppAbility>: TS2345: Argument of type  is not assignable to parameter of type

### DISCUSSION THREAD

**Discussion by xtcodecom** [stalniy/casl#961] [Q&A]
```
import { MongoAbility, RawRuleOf } from '@casl/ability';

type Abilities = ['read', 'User'];
type AppAbility = MongoAbility<Abilities>;

const rule: RawRuleOf<AppAbility> = { action: 'read', subject: 'User' };

ability.can(rule.action, rule.subject);
```

Getting TS error: 

> TS2345: Argument of type ['read' | 'read'[], 'User' | 'User'[]] is not assignable to parameter of type

for ability.can(...) function.
What is the correct rule to be used here?

**Comment by stalniy** [comment]
You can't use it like this because `RawRuleOf<AppAbility>['action']` is not only `read` it's `read | read[]` type literal. That's why TS complains about it. What you need to do is:

```ts
const rule = { action: 'read', subject: 'User' } satisfies RawRuleOf<AppAbility> ;

ability.can(rule.action, rule.subject);
```

or for older TS versions:

```ts
const rule = { action: 'read', subject: 'User' } as const;

ability.can(rule.action, rule.subject);
```
